bot/cogs/doc/cache.py

```diff
@@ -0,0 +1,32 @@
+import functools
+from collections import OrderedDict
+from typing import Any, Callable
+
+
+def async_cache(max_size: int = 128, arg_offset: int = 0) -> Callable:
```

Comment:
This is pretty generic and may be useful elsewhere in the future. It can go in decorators.py

Reply:
I noticed you clear the cache sometimes, so this won't work nicely if other things use it too. Well, you can keep it here for now and if we find a use for it later we'll refactor it.

Reply:
#901 Should make a more reusable async cache so I avoided making too much here and only moved it in the meantime